src/config/index.ts

```diff
@@ -186,6 +191,33 @@ export class Config {
     return _.sortBy(domains, 'friendly');
   }
 
+  // TODO: Joi? Chai?
+  public static assertIfInvalid({ config }: PartnerConfig = partnerConfig) {
+    for (const contactType of config.contact_types) {
+      const allHierarchyProperties = [...contactType.hierarchy, contactType.replacement_property];
+      const allProperties = [
+        ...contactType.place_properties,
+        ...contactType.contact_properties,
+        ...allHierarchyProperties,
+        Config.getUserRoleConfig(contactType),
+      ];
+      
+      Config.getPropertyWithName(contactType.place_properties, 'name');
```

Comment:
true. now that we assert it on startup, maybe we don't need it in the property? we know it is also there

Reply:
> maybe we don't need it in the property

Yeah i think it would be clearer if `getPropertyWithName` returned a string or undefined; then the caller can choose to assert if undefined and throw if needed

Reply:
since we assert at startup, we can guarantee it is never undefined at runtime
this allows us to safely make it a `string` which takes the burden off the calling code
i personally like this since it leads to cleaner calling code.  would you rather i not assert in the get and just use typescript operators to cast?

Reply:
Wouldn't we get a runtime error if the property name that was being accessed was not asserted at start up like `name` is? If this function is only used for getting `name` then maybe it can be renamed and args changed?